src/connectors/refinement-list/connectRefinementList.js

```diff
@@ -375,13 +388,14 @@ export default function connectRefinementList(renderFn, unmountFn = noop) {
       dispose({ state }) {
         unmountFn();
 
+        let newState;
         if (operator === 'and') {
-          return state.removeFacetRefinement(attribute).removeFacet(attribute);
+          newState = state.removeFacet(attribute);
         } else {
-          return state
-            .removeDisjunctiveFacetRefinement(attribute)
-            .removeDisjunctiveFacet(attribute);
+          newState = state.removeDisjunctiveFacet(attribute);
         }
+
+        return newState.setQueryParameter('maxValuesPerFacet', undefined);
```

Comment:
we need to make sure we only  unset maxValuesPerFacet if it was set by this widget

Reply:
decided that it's ok, because the other mounted widgets will set their value back